spring-xd-module/src/main/java/org/springframework/xd/module/ModuleDescriptor.java

```diff
@@ -575,6 +575,9 @@ public Builder setModuleDefinition(ModuleDefinition moduleDefinition) {
 		}
 
 		public ModuleDefinition getModuleDefinition() {
+			if (moduleDefinition == null) {
+				moduleDefinition = ModuleDefinitions.dummy(moduleName, type);
```

Comment:
`ModuleDefinitions.dummy()` itself should be removed and replaced with something that is only on the src/test side. The fact that something "should not be used" is an indicator that it's in the wrong place.